enhancements/network/multi-networkpolicy.md

```diff
@@ -0,0 +1,178 @@
+---
+title: multi-networkpolicy
+authors:
+  - "@s1061123"
+reviewers:
+  - @dcbw
+  - @dougbtv
+  - @fepan
+  - @zshi-redhat
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-08-07
+last-updated: yyyy-mm-dd
+status: implementable
+---
+
+# NetworkPolicy for Multus Interfaces
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This proposal to bring networkpolicy features for multus interface.
+
+## Motivation
+
+Customer wants to make their Kubernetes secure, by filtering traffic from
+unexpected source. Kubernetes provides `NetworkPolicy` feature, however, it
+is only applied to default network interface, not additional network interface
+that is created by Multus CNI.  NetworkPolicy for Multus Interfaces brings the NetworkPolicy
+mechanism to additional network interfaces and it provides customers the ability to filter
+traffic from unexpected sources.
+
+### Goals
+
+- Introduce MultiNetworkPolicy CRD
+- Introduce multi-networkpolicy which generates iptables from CRD as shown in the Design Details section
+- Provide iptable based filtering mechanism for pod
+- Gathering related information (iptables rules in pod network namespace) by must-gather
+
+### Non-Goals
+
+- Provide eBPF filtering mechanism
+- Provide NetworkPolicy for userspace pod which uses DPDK and so on
+
+## Proposal
+
+### User Stories [optional]
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+#### Story 1
+
+Customer can specify how groups of pods are allowed to communicate with each other
+and other network endpoints as Kubernetes NetworkPolicy does.
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Multus
+
+No additional change is required because multi-networkpolicy only consumes 
+network-attachment-definition CRD and it does not change net-attach-def CRD.
+
+### How user enables multi-networkpolicy?
+
+Currently we're going to bring it as Tech Preview first and then GA after a few release,
+hence opt-in style configuration is planned at Tech Preview and make it default at GA.
+Opt-in style configuration will be done at cluster-network-operator as boolean flag
+(such as `useMultiNetworkPolicy: true`).
```

Comment:
I suppose we should have it because NetworkPolicy controller is not mandatory component for OCP (although it is GA in the future). We should have explicit way to enable/disable it for administrator.